File: javascript/bubbleSort.js
function bubble_sort(arr) {
  let unsorted_arr_idx = arr.length - 1;
  let sorted = false;

  while (!sorted) {
    sorted = true;
    for (let i = 0; i < unsorted_arr_idx; i++) {
      sorted = false;
      const a = arr[i];
      const b = arr[i + 1];
      arr[i] = b;
      arr[i + 1] = a;
      // arr[i + 1] = arr[i];
      // console.log(arr[i], arr[i + 1]);
    }
    unsorted_arr_idx = unsorted_arr_idx - 1;
  }
}

arr = [65, 55, 45, 35, 25, 15, 10];
bubble_sort(arr);
console.log(arr);

File: python/bubbleSort.py
def bubble_sort(li):
    unsorted_li_idx = len(li) - 1
    sorted = False

    while not sorted:
        sorted = True
        for i in range(unsorted_li_idx):
            print(i,i+1)
            if li[i] > li[i+1]:
                sorted = False
                li[i], li[i+1] = li[i+1], li[i]
        unsorted_li_idx = unsorted_li_idx -1

li = [65,55,45,35,25,15,10]
bubble_sort(li)
print(li)